### ISSUE TITLE
matplotlib-2.1.0 bug (Anaconda 5.0.1) with interactive mode figure() on macOS

### ISSUE THREAD

**Issue by tloredo** [ContinuumIO/anaconda-issues#7084]
After upgrading Anaconda from 5.0.0 (with matplotlib-2.0.2) to 5.0.1 (with matplotlib-2.1.0), I get a fatal error every time I try to open a figure in IPython interactive mode:

```
In [1]: figure()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-097dcd9e934c> in <module>()
----> 1 figure()

/Users/loredo/anaconda/lib/python2.7/site-packages/matplotlib/pyplot.pyc in figure(num, figsize, dpi, facecolor, edgecolor, frameon, FigureClass, clear, **kwargs)
    554         # FigureManager base class.
    555         if matplotlib.is_interactive():
--> 556             draw_if_interactive()
    557
    558         if _INSTALL_FIG_OBSERVER:

/Users/loredo/anaconda/lib/python2.7/site-packages/IPython/utils/decorators.pyc in wrapper(*args, **kw)
     41     def wrapper(*args,**kw):
     42         wrapper.called = False
---> 43         out = func(*args,**kw)
     44         wrapper.called = True
     45         return out

/Users/loredo/anaconda/lib/python2.7/site-packages/matplotlib/backend_bases.pyc in draw_if_interactive(cls)
    183             manager = Gcf.get_active()
    184             if manager:
--> 185                 cls.trigger_manager_draw(manager)
    186
    187     @classmethod

TypeError: unbound method trigger_manager_draw() must be called with _BackendMac instance as first argument (got FigureManagerMac instance instead)
```

It appears this is a known bug for macOS with mpl-2.1.0 and interactive mode:

*  [python - matplotlib broken after `pip install matplotlib --upgrade` to version 2.1.0 - Ask Different](https://apple.stackexchange.com/questions/301961/matplotlib-broken-after-pip-install-matplotlib-upgrade-to-version-2-1-0)
* [matplotlib 2.1.0, backend macosx: need _BackendMac, got FigureManagerMac · Issue #9345 · matplotlib/matplotlib](https://github.com/matplotlib/matplotlib/issues/9345)

The recommended solution is to not use interactive mode, or to downgrade. I did the latter (`conda install matplotlib=2.0.2`); it appears there are no dependencies and only mpl was replaced. That said, mpl-2.1.0 seems to me to be an unfortunate choice for anaconda-5.0.1 as this is a pretty fundamental bug. I hope there will be an Anaconda update as soon as this is fixed.

**Comment by mingwandroid** [comment]
Did you try the workaround here: https://github.com/matplotlib/matplotlib/issues/9345#issuecomment-341197456

**Comment by tloredo** [comment]
I did see it, but I'm not sure I'd consider turning off capability I want a real workaround.  8-) Esp. since I'm preparing material for a course next semester, I want to use a standard environment that just works with the normal interactive mode, and without having to make students use a workaround. Also, re: the pylab part of that post, I don't ever use pylab. Thanks for highlighting it, though; could be useful for others seeing this issue and wanting to stick with 2.1.0.

**Comment by mingwandroid** [comment]
I do not think we will revert an updated version, it is essential that upstream fixes this before we can do anything. @csoja, making you aware of this request.

**Comment by tloredo** [comment]
Thanks for the ping. I wasn't suggesting a reversion in anaconda; I just mentioned that reversion works without impacting other packages as potentially useful info for others with the problem. I just wanted to call attention to it in the hope that an upstream fix would quickly trigger an anaconda update.

**Comment by mingwandroid** [comment]
Yup, thanks for the ping. Anaconda releases are not something that can be done 'quickly', they take *significant* effort and require a prolonged QA period.

**Comment by tloredo** [comment]
Got it, and makes sense. Testing IPython interactive mode should be added to the QA. 8-)

**Comment by mingwandroid** [comment]
I know that's semi in-jest, but we support so many use-cases and users that it's very difficult. We do run the testsuite for every package though, pretty much (depending on what the recipe has in its `tests` section).

.. still if it exercises a good amount of the stack then I'm for it. Care to share a minimal script that exhibits the bug?

**Comment by hopehhchen** [comment]
I propose to close this issue.  It can now be easily resolved by updating matplotlib to 2.2+.

**Comment by tloredo** [comment]
Agreed this can be closed; it's no longer a problem with a current Anaconda, and was fixed before my Spring 2018 students would have encountered it.
